web/src/app/users/UserContactMethodCreateDialog.js

```diff
@@ -17,68 +16,54 @@ const createMutation = gql`
   }
 `
 
-export default class UserContactMethodCreateDialog extends React.PureComponent {
-  static propTypes = {
-    userID: p.string.isRequired,
-    onClose: p.func,
-  }
-
-  state = {
-    value: {
-      name: '',
-      type: 'SMS',
-      value: '+1',
+export default function UserContactMethodCreateDialog(props) {
+  // values for contact method form
+  const [cmValue, setCmValue] = useState({
+    name: '',
+    type: 'SMS',
+    value: '+1',
+  })
+  const [createCM, createCMStatus] = useMutation(createMutation, {
+    refetchQueries: ['nrList', 'cmList'],
+    awaitRefetchQueries: true,
+    onCompleted: result => {
+      props.onClose({ contactMethodID: result.createUserContactMethod.id })
     },
-    errors: [],
-  }
-
-  render() {
-    return (
-      <Mutation
-        client={graphql2Client}
-        mutation={createMutation}
-        awaitRefetchQueries
-        refetchQueries={['nrList', 'cmList']}
-        onCompleted={this.props.onClose}
-      >
-        {(commit, status) => this.renderDialog(commit, status)}
-      </Mutation>
-    )
-  }
-
-  renderDialog(commit, status) {
-    const { loading, error } = status
+    variables: {
+      input: {
+        ...cmValue,
+        userID: props.userID,
+        newUserNotificationRule: {
+          delayMinutes: 0,
+        },
+      },
+    },
+  })
 
-    const fieldErrs = fieldErrors(error)
+  const { loading, error } = createCMStatus
+  const fieldErrs = fieldErrors(error)
+  const form = (
+    <UserContactMethodForm
+      disabled={loading}
+      errors={fieldErrs}
+      onChange={cmValue => setCmValue(cmValue)}
+      value={cmValue}
+    />
+  )
+  return (
+    <FormDialog
+      data-cy='create-form'
+      title='Create New Contact Method'
+      loading={loading}
+      errors={nonFieldErrors(error)}
+      onClose={props.onClose}
+      onSubmit={() => createCM()}
```

Comment:
you can reference `createCm` here without the wrapper function:

`onSubmit={createCM}`

Reply:
Won't that pass the `onSubmit` event handler arguments as options to the mutation function?

Reply:
*Update* Discussed offline, let's go back to the wrapper, but add a comment noting why (we don't want the SyntheticEvent being passed to `createCM` or something to that effect).